src/openrct2/platform/Windows.cpp

```diff
@@ -222,15 +221,14 @@ bool platform_get_steam_path(utf8* outPath, size_t outSize)
         return false;
     }
 
-    wSteamPath = (wchar_t*)malloc(size);
-    result = RegQueryValueExW(hKey, L"SteamPath", nullptr, &type, (LPBYTE)wSteamPath, &size);
+    std::unique_ptr<wchar_t> wSteamPath(new wchar_t[size]);
```

Comment:
This is creating a `std::unique_ptr<wchar_t>` which will call `delete` on the destructor. But the resource being handled is an array, so `delete[]` should be called.
Hence the correct syntax would be `auto wSteamPath = std::make_unique<wchar_t[]>(size)`